spring-shell2-shell1-adapter/src/main/java/org/springframework/shell2/legacy/LegacyParameterResolver.java

```diff
@@ -82,7 +91,58 @@ public Object resolve(MethodParameter methodParameter, List<String> words) {
 
 	@Override
 	public ParameterDescription describe(MethodParameter parameter) {
-		throw new UnsupportedOperationException();
+		Parameter jlrParameter = parameter.getMethod().getParameters()[parameter.getParameterIndex()];
+		CliOption option = jlrParameter.getAnnotation(CliOption.class);
+		ParameterDescription result = ParameterDescription.outOf(parameter);
+		result.help(option.help());
+		List<String> keys = Arrays.asList(option.key());
+		result.keys(keys.stream()
+				.filter(key -> !key.isEmpty())
+				.map(key -> CLI_PREFIX + key)
+				.collect(Collectors.toList()));
+		Optional<String> defaultValue = defaultValueFor(option, result.keys());
+		if (defaultValue.isPresent()) {
+			result.defaultValue(defaultValue.get());
+		}
+		boolean containsEmptyKey = keys.contains("");
+		result.mandatoryKey(!containsEmptyKey);
```

Comment:
Indeed key being mandatory should only be related to whether or not it is [an alias of] the "default" key (ie `""`).

> Can a mandatory CliOption have an unspecifiedDefaultValue? 

maybe

> (or better put, should it be allowed?)

no :)

I think we're good too